Code/Tools/Libs/ToolsFoundation/NodeObject/DocumentNodeManager.h

```diff
@@ -123,84 +126,56 @@ class EZ_TOOLSFOUNDATION_DLL ezDocumentNodeManager : public ezDocumentObjectMana
   };
 
   bool IsNode(const ezDocumentObject* pObject) const;
-  ezStatus CanConnect(const ezPin* pSource, const ezPin* pTarget, CanConnectResult& result) const;
+  bool IsConnection(const ezDocumentObject* pObject) const;
+
+  ezArrayPtr<const ezConnection* const> GetConnections(const ezPin& pin) const;
+  bool HasConnections(const ezPin& pin) const;
+  bool IsConnected(const ezPin& source, const ezPin& target) const;
+  ezStatus CanConnect(const ezPin& source, const ezPin& target, CanConnectResult& result) const;
```

Comment:
Nit: I would suggest making the CanConnect and Connect functions roughly have the same signature. In this case, you ommit the pObject so it could be that later an unsupported type is thrown in. Similiar to CanAdd on the object manager, you could put in `const ezRTTI* pObjectType` to identify the type of connection you want to add. I know in this case it is not really necessary as every document type currently has one fixed connection type so the logic is rather uninteresting.

Reply:
Done